fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -352,7 +352,7 @@ def _clear_counters(self) -> None:
             assert self._bucket_list is not None
 
             for bucket in self._bucket_list:
-                assert self.should_accumulate_grads or bucket.sent, (
+                assert not self.training or self.should_accumulate_grads or bucket.sent, (
                     "A bucket failed to be sent, probably unused parameters."
```

Comment:
Should we update this message?

Reply:
I think it's ok right now, basically this assert would throw before when in evaluation mode, because the .eval() call was not properly caught